lib/content-services/src/lib/search/components/search-date-range-advanced-tabbed/search-date-range-advanced-tabbed.component.spec.ts

```diff
@@ -0,0 +1,215 @@
+       fixture.detectChanges();
+       expect(component.fields.length).toBe(2);
+       expect(component.fields).toEqual(['createdDate', 'modifiedDate']);
+    });
+
+    it('should return hasValidValue as false if any of the fields has an invalid value', () => {
+       component.tabsValidity['createdDate'] = false;
+       fixture.detectChanges();
+       expect(component.hasValidValue()).toBeFalse();
+       fixture.detectChanges();
+       component.tabsValidity['modifiedDate'] = false;
+       fixture.detectChanges();
+       expect(component.hasValidValue()).toBeFalse();
+       component.tabsValidity['createdDate'] = true;
+       component.tabsValidity['modifiedDate'] = true;
+       fixture.detectChanges();
+       expect(component.hasValidValue()).toBeTrue();
+    });
+
+    it('should update displayValue when values are submitted', () => {
+        spyOn(component.displayValue$, 'next');
+        component.onDateRangedValueChanged(betweenMockData, 'createdDate');
+        component.onDateRangedValueChanged(inLastMockData, 'modifiedDate');
+        fixture.detectChanges();
+        component.submitValues();
+        expect(component.displayValue$.next).toHaveBeenCalledWith('CREATED DATE: 05-Jun-23 - 07-Jun-23 MODIFIED DATE: SEARCH.DATE_RANGE_ADVANCED.IN_LAST_DISPLAY_LABELS.WEEKS');
+
+        component.onDateRangedValueChanged(anyMockDate, 'createdDate');
+        component.onDateRangedValueChanged(anyMockDate, 'modifiedDate');
+        fixture.detectChanges();
+        component.submitValues();
+        expect(component.displayValue$.next).toHaveBeenCalledWith('');
+    });
+
+    it('should update query when values are changed', () => {
+        component.onDateRangedValueChanged(betweenMockData, 'createdDate');
+        component.onDateRangedValueChanged(inLastMockData, 'modifiedDate');
+        fixture.detectChanges();
+        let query = `createdDate:['${formatISO(startOfDay(betweenMockData.betweenStartDate))}' TO '${formatISO(endOfDay(betweenMockData.betweenEndDate))}'] AND modifiedDate:[NOW/DAY-5WEEKS TO NOW/DAY+1DAY]`;
+        expect(component.combinedQuery).toEqual(query);
+
+        inLastMockData = {
+            dateRangeType: DateRangeType.IN_LAST,
+            inLastValueType: InLastDateType.DAYS,
+            inLastValue: '9',
+            betweenStartDate: null,
+            betweenEndDate: null
+        }
+        component.onDateRangedValueChanged(inLastMockData, 'modifiedDate');
+        fixture.detectChanges();
+        query = `createdDate:['${formatISO(startOfDay(betweenMockData.betweenStartDate))}' TO '${formatISO(endOfDay(betweenMockData.betweenEndDate))}'] AND modifiedDate:[NOW/DAY-9DAYS TO NOW/DAY+1DAY]`;
+        expect(component.combinedQuery).toEqual(query);
+
+        inLastMockData = {
+            dateRangeType: DateRangeType.IN_LAST,
+            inLastValueType: InLastDateType.MONTHS,
+            inLastValue: '7',
+            betweenStartDate: null,
+            betweenEndDate: null
+        }
```

Comment:
missing semicolon

Reply:
These has been resolved in [this](https://github.com/Alfresco/alfresco-ng2-components/pull/8714/commits/bdaa0e52fd3a31ba7c3802c6cdc6a27a733d6057) commit